detox/android/detox/src/main/java/com/wix/detox/DetoxJUnitRunner.java

```diff
@@ -34,12 +34,25 @@ public void onApplicationLifecycleChanged(Application app, ApplicationStage stag
         monitor.addLifecycleCallback(lifecycleCallback);
     }
 
+    @Override
+    public void onDestroy() {
+        instrumentsManager.stopRecording();
+        instrumentsManager = null;
+        lifecycleCallback = null;
+
+        super.onDestroy();
+    }
```

Comment:
?? ❤️

Reply:
This one was missed before. So better cleaning for self late than never.

Reply:
> ?? ❤️

This one was missed before. So better cleaning for self late than never.